airflow/providers/amazon/aws/hooks/kinesis.py

```diff
@@ -44,3 +44,41 @@ def __init__(self, delivery_stream: str, *args, **kwargs) -> None:
     def put_records(self, records: Iterable):
         """Write batch records to Kinesis Firehose"""
         return self.get_conn().put_record_batch(DeliveryStreamName=self.delivery_stream, Records=records)
+
+
+class KinesisHook(AwsBaseHook):
+    """
+    Interact with AWS Kinesis Data Stream.
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    :param delivery_stream: Name of the data stream
+
+    AWS Kinesis docs: https://docs.aws.amazon.com/streams/latest/dev/introduction.html
+
+    .. seealso::
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+    """
+
+    def __init__(self, delivery_stream: str, *args, **kwargs) -> None:
+        self.delivery_stream = delivery_stream
+        kwargs["client_type"] = "kinesis"
+        super().__init__(*args, **kwargs)
+
+    def put_records(self, records: list[dict[str, str| bytes]]):
```

Comment:
This method does not seem to add any value to [boto3.put_records](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/kinesis.html#Kinesis.Client.put_record) method. It is a just a wrapper around it. We are trying to avoid such method, it adds a layer for no additional value. I know this how it was done in `FirehoseHook` but in my opinion we should not follow this pattern. A good example is the hook `/airflow/providers/amazon/aws/hooks/rds.py` which provides nothing but the connection object to query `boto3`

Reply:
Yes! completely agree!  All the autocomplete, and none of the code!

Reply:
Just realized that's what you exactly said in your previous comment (oops)